cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -140,23 +141,74 @@ fn calculate_soroban_resources(
     })
 }
 
-fn get_fee_configuration(_snapshot_source: &ledger_storage::LedgerStorage) -> FeeConfiguration {
-    // TODO: (at least part of) these values should be obtained from the network's ConfigSetting LedgerEntries
-    //       (instead of hardcoding them to the initial values in the network)
-    //       Specifically, we need to derive it from ConfigSettingContractComputeV0 which can
-    //       be retrieved using the ConfigSetting/CONFIG_SETTING_CONTRACT_COMPUTE_V0.
+fn get_configuration_setting(
+    ledger_storage: &ledger_storage::LedgerStorage,
+    setting_id: ConfigSettingId,
+) -> Result<ConfigSettingEntry, Box<dyn error::Error>> {
+    let key = LedgerKey::ConfigSetting(LedgerKeyConfigSetting {
+        config_setting_id: setting_id,
+    });
+    match ledger_storage.get(&key)? {
+        LedgerEntry {
+            data: LedgerEntryData::ConfigSetting(cs),
+            ..
+        } => Ok(cs),
+        _ => Err(format!(
+            "get_configuration_setting(): unexpected ledger entry for {} key",
+            setting_id.name()
+        )
+        .into()),
+    }
+}
+
+fn get_fee_configuration(
+    ledger_storage: &ledger_storage::LedgerStorage,
+) -> Result<FeeConfiguration, Box<dyn error::Error>> {
+    // TODO: consider caching these entries. In theory, they can change at any given ledger
+    //       (making caching ineffective) but we can maybe relax that requirement?
```

Comment:
If they don't change frequently, but we need immediate results when they *do* change, a write-through cache might be the right choice here.

Reply:
That's a really good idea

Reply:
I will add a comment about it